shared/test/unit/mocks/mock_service.js

```diff
@@ -28,8 +48,52 @@ var MockService = {
   },
   query: function(state) {
     switch (state) {
+      case 'isUpgrade':
```

Comment:
I have no long term plan... but I am opened to other solution to avoid put instances under global object.